src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,31 +23,59 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
-
-        public ConfigurationClientManager(IEnumerable<string> connectionStrings, ConfigurationClientOptions clientOptions)
-        {
-            if (connectionStrings == null || !connectionStrings.Any())
-            {
-                throw new ArgumentNullException(nameof(connectionStrings));
-            }
-
-            _clients = connectionStrings.Select(connectionString => 
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
-            if (endpoints == null || !endpoints.Any())
-            {
-                throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
+    {
+        private IList<ConfigurationClientWrapper> _clients;
+
+        private IEnumerable<string> _connectionStrings;
+        private IEnumerable<Uri> _endpoints;
+        private TokenCredential _credential;
+        private ConfigurationClientOptions _clientOptions;
+        private Lazy<SrvLookupClient> _srvLookupClient;
+
+        public ConfigurationClientManager(AzureAppConfigurationOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (options.ClientOptions == null)
+            {
+                throw new ArgumentException(nameof(options.ClientOptions));
+            }
+
+            _connectionStrings = options.ConnectionStrings;
+            _endpoints = options.Endpoints;
+            _credential = options.Credential;
+            _clientOptions = options.ClientOptions;
+
+            _srvLookupClient = new Lazy<SrvLookupClient>();
+
+            if (_connectionStrings != null && _connectionStrings.Any())
+            {
+                _clients = _connectionStrings
+                    .Select(connectionString =>
+                        {
+                            var endpoint = new Uri(ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.EndpointSection));
+                            return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, _clientOptions));
+                        })
+                    .ToList();
+            }
+            else if (_endpoints != null && _endpoints.Any())
+            {
+                if (_credential == null)
+                {
+                    throw new ArgumentNullException(nameof(options.Credential));

```

Comment:
ArgumentException is more appropriate, note that it's fine in some cases to pass credential as null. What's actually invalid here is the options argument because it doesn't have any connection strings and it also doesn't have a credential.